### ISSUE TITLE
Argo waiter: invalid memory address or nil pointer dereference with port-forward

### ISSUE THREAD

**Issue by Siebjee** [argoproj/argo-workflows#9336]
## Summary

When a `argo wait <workflow>` is used via `kubectl port-forward svc/argo-server 8443:2746 -n argo` it eventually panics with a `invalid memory address or nil pointer dereference`.

```
Error: runtime error: invalid memory address or nil pointer dereference
```

What version are you running?

3.3.8

## Diagnostics

Paste the smallest workflow that reproduces the bug. We must be able to run the workflow.

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
    name: argo

---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: argo-wf-50-binding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argo-cluster-role
subjects:
- kind: ServiceAccount
  name: argo
  namespace: default

---
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: test-sleeper-50
spec:
  entrypoint: pipeline
  serviceAccountName: argo
  templates:
  - name: pipeline
    steps:
    - - name: sleeper
        template: sleeper

  - name: sleeper
    container:
      image: alpine
      command: [sh, -c]
      env:
      - name: SEC
        value: "3600"
      args:
      - |
        for i in `seq 1 ${SEC}`; do
          printf "%4d / %4d\n" $i ${SEC}
          sleep 1
        done
```

### Reproduction steps:

**Please note, here i use `argo-ui` as this is the service account you have access to in our argo instance. Please adjust accordingly.**

```bash
export ARGO_TOKEN="Bearer $(kubectl -n argo get secret $(kubectl -n argo get sa argo-ui -o=jsonpath='{.secrets[0].name}') -o=jsonpath='{.data.token}' | base64 --decode)"

kubectl port-forward svc/argo-server 8443:2746 -n argo &
kubectl apply -f <above manifest> -n default

ARGO_SERVER=localhost:8443 ARGO_HTTP1=true ARGO_SECURE=true ARGO_INSECURE_SKIP_VERIFY=true /usr/local/bin/argo wait test-sleeper-50
```
Somewhere after 30 minutes, before 50 minutes the `argo waiter` should have panic'd.

Logs below show namespace `siebren` because i ran everything in that namespace. Example i converted this all to `default`
```bash
# Logs from the workflow controller:
kubectl logs -n argo deploy/workflow-controller | grep ${workflow} 

time="2022-08-11T08:51:31.611Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.616Z" level=info msg="Updated phase  -> Running" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.617Z" level=info msg="Steps node test-sleeper-50 initialized Running" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.617Z" level=info msg="StepGroup node test-sleeper-50-2995923228 initialized Running" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.617Z" level=info msg="Pod node test-sleeper-50-3381761298 initialized Pending" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.647Z" level=info msg="Created pod: test-sleeper-50[0].sleeper (test-sleeper-50-3381761298)" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.647Z" level=info msg="Workflow step group node test-sleeper-50-2995923228 not yet completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.647Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.647Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:31.659Z" level=info msg="Workflow update successful" namespace=siebren phase=Running resourceVersion=175660953 workflow=test-sleeper-50
time="2022-08-11T08:51:41.649Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:41.649Z" level=info msg="Task-result reconciliation" namespace=siebren numObjs=0 workflow=test-sleeper-50
time="2022-08-11T08:51:41.649Z" level=info msg="node changed" new.message= new.phase=Running new.progress=0/1 nodeID=test-sleeper-50-3381761298 old.message= old.phase=Pending old.progress=0/1
time="2022-08-11T08:51:41.650Z" level=info msg="Workflow step group node test-sleeper-50-2995923228 not yet completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:41.650Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:41.650Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:41.659Z" level=info msg="Workflow update successful" namespace=siebren phase=Running resourceVersion=175661114 workflow=test-sleeper-50
time="2022-08-11T08:51:51.660Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:51.661Z" level=info msg="Task-result reconciliation" namespace=siebren numObjs=0 workflow=test-sleeper-50
time="2022-08-11T08:51:51.661Z" level=info msg="node unchanged" nodeID=test-sleeper-50-3381761298
time="2022-08-11T08:51:51.661Z" level=info msg="Workflow step group node test-sleeper-50-2995923228 not yet completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:51.661Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T08:51:51.661Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:11:51.661Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:11:51.661Z" level=info msg="Task-result reconciliation" namespace=siebren numObjs=0 workflow=test-sleeper-50
time="2022-08-11T09:11:51.661Z" level=info msg="node unchanged" nodeID=test-sleeper-50-3381761298
time="2022-08-11T09:11:51.661Z" level=info msg="Workflow step group node test-sleeper-50-2995923228 not yet completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:11:51.661Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:11:51.661Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:31:51.661Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:31:51.662Z" level=info msg="Task-result reconciliation" namespace=siebren numObjs=0 workflow=test-sleeper-50
time="2022-08-11T09:31:51.662Z" level=info msg="node unchanged" nodeID=test-sleeper-50-3381761298
time="2022-08-11T09:31:51.662Z" level=info msg="Workflow step group node test-sleeper-50-2995923228 not yet completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:31:51.662Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:31:51.662Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.294Z" level=info msg="Processing workflow" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.295Z" level=info msg="Task-result reconciliation" namespace=siebren numObjs=0 workflow=test-sleeper-50
time="2022-08-11T09:51:40.295Z" level=warning msg="workflow uses legacy/insecure pod patch, see https://argoproj.github.io/argo-workflows/workflow-rbac/" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.295Z" level=info msg="node changed" new.message="Pod was active on the node longer than the specified deadline" new.phase=Failed new.progress=0/1 nodeID=test-sleeper-50-3381761298 old.message= old.phase=Running old.progress=0/1
time="2022-08-11T09:51:40.296Z" level=info msg="Step group node test-sleeper-50-2995923228 deemed failed: child 'test-sleeper-50-3381761298' failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50-2995923228 phase Running -> Failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50-2995923228 message: child 'test-sleeper-50-3381761298' failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50-2995923228 finished: 2022-08-11 09:51:40.296067247 +0000 UTC" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="step group test-sleeper-50-2995923228 was unsuccessful: child 'test-sleeper-50-3381761298' failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Outbound nodes of test-sleeper-50-3381761298 is [test-sleeper-50-3381761298]" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Outbound nodes of test-sleeper-50 is [test-sleeper-50-3381761298]" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50 phase Running -> Failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50 message: child 'test-sleeper-50-3381761298' failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="node test-sleeper-50 finished: 2022-08-11 09:51:40.296135588 +0000 UTC" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Checking daemoned children of test-sleeper-50" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="TaskSet Reconciliation" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg=reconcileAgentPod namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Updated phase Running -> Failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Updated message  -> child 'test-sleeper-50-3381761298' failed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Marking workflow completed" namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.296Z" level=info msg="Checking daemoned children of " namespace=siebren workflow=test-sleeper-50
time="2022-08-11T09:51:40.301Z" level=info msg="cleaning up pod" action=deletePod key=siebren/test-sleeper-50-1340600742-agent/deletePod
time="2022-08-11T09:51:40.306Z" level=info msg="Workflow update successful" namespace=siebren phase=Failed resourceVersion=175705798 workflow=test-sleeper-50
time="2022-08-11T09:51:40.306Z" level=info msg="Queueing Failed workflow siebren/test-sleeper-50 for delete in 24h0m0s due to TTL"
time="2022-08-11T09:51:45.312Z" level=info msg="cleaning up pod" action=deletePod key=siebren/test-sleeper-50-3381761298/deletePod
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
Could you try the latest version v3.4-rc1?

**Comment by Siebjee** [comment]
I shall Monday 👍🏼, have a busy weekend ahead of me.

**Comment by Siebjee** [comment]
Sorry for the delay @terrytangyuan , But i hit the same error with `3.4.0-rc1`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x358 pc=0x280aff5]

goroutine 12 [running]:
github.com/argoproj/argo-workflows/v3/cmd/argo/commands/common.waitOnOne({0x42d0ee8, 0xc0003cbb00}, {0x42c2818, 0xc000138008}, {0x7ff7bfefeebd, 0xf}, {0x7ff7bfefeeb0, 0x7}, 0x0, 0x0)
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/commands/common/wait.go:72 +0x2d5
github.com/argoproj/argo-workflows/v3/cmd/argo/commands/common.WaitWorkflows.func1({0x7ff7bfefeebd?, 0x0?})
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/commands/common/wait.go:29 +0x78
created by github.com/argoproj/argo-workflows/v3/cmd/argo/commands/common.WaitWorkflows
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/commands/common/wait.go:28 +0x26d

```

**Comment by Siebjee** [comment]
Confirm fixed with release [3.4.0-rc3](https://github.com/argoproj/argo-workflows/releases/tag/v3.4.0-rc3)
